common/src/main/kotlin/org/waste/of/time/storage/cache/HotCache.kt

```diff
@@ -65,10 +67,19 @@ object HotCache {
      *
      * @param chunkX The X coordinate of the chunk.
      * @param chunkZ The Z coordinate of the chunk.
+     * @param dimension The dimension of the chunk.
      * @return True if the chunk is saved, false otherwise.
      */
     @Suppress("unused")
-    fun isChunkSaved(chunkX: Int, chunkZ: Int) = savedChunks.contains(ChunkPos.toLong(chunkX, chunkZ))
+    fun isChunkSaved(chunkX: Int, chunkZ: Int, dimension: RegistryKey<World>): Boolean {
+        val dimensionChunks = savedChunks[dimension] ?: return false
+        return dimensionChunks.contains(ChunkPos.toLong(chunkX, chunkZ))
+    }
+
+    @Suppress("unused")
+    fun getSavedChunks(dimension: RegistryKey<World>): LongArrayList {
```

Comment:
change return type to more generic `LongCollection` here so underlying type can be changed as needed